tests/e2e/encrypted.sh

```diff
@@ -0,0 +1,182 @@
+	if [ -d "${guest_components_path}" ]; then
+		echo "Guest components repo directory exists, removing it"
+		rm -rf "${guest_components_path}"
+	fi
+
+	git clone "${guest_components_repo}"
+
+	pushd "${guest_components_path}"
+  		export tag_name="coco-keyprovider"
+ 		docker build -t "${tag_name}" -f ./attestation-agent/docker/Dockerfile.keyprovider .
+   		mkdir -p oci/{input,output}
+   		skopeo copy docker-daemon:unencrypted:latest dir:./oci/input
+   		docker run -v "${PWD}/oci:/oci" "${tag_name}" /encrypt.sh -k "${key_b64}" -i "${key_id}" -s dir:/oci/input -d dir:/oci/output
+   		skopeo inspect dir:./oci/output | jq '.LayersData[0].Annotations["org.opencontainers.image.enc.keys.provider.attestation-agent"] | @base64d | fromjson'
+ 		skopeo copy dir:./oci/output "docker://${encrypted_image}"
+		skopeo inspect "docker://${encrypted_image}" | jq -r '.Digest'
+		# Verify layer is encrypted
+		skopeo inspect "docker://${encrypted_image}" | jq -r '.LayersData[].MIMEType' | grep encrypted
+		skopeo inspect "docker://${encrypted_image}" | jq -r '.LayersData[].Annotations."org.opencontainers.image.enc.keys.provider.attestation-agent"' | base64 -d
+		skopeo inspect "docker://${encrypted_image}" | jq -r '.LayersData[].Annotations."org.opencontainers.image.enc.pubopts"' | base64 -d
+	popd
+}
+
+deploy_k8s_kbs() {
+	if [ ! -d "${script_dir}/trustee" ]; then
+		git clone "${trustee_repo}"
+	fi
+
+	pushd "${script_dir}/trustee/kbs/config/kubernetes"
+		echo "somesecret" > overlays/$(uname -m)/key.bin
+		export DEPLOYMENT_DIR=nodeport
+		./deploy-kbs.sh
+		kbs_pod=$(kubectl -n "${kbs_namespace}" get pods -o NAME)
+		kubectl wait --for=condition=Ready -n "${kbs_namespace}" "${kbs_pod}" --timeout=300s
+	popd
+}
+
+delete_k8s_kbs() {
+	pushd "${script_dir}/trustee/kbs/config/kubernetes"
+		kubectl delete -k overlays/$(uname -m)
+	popd
+	rm -rf "${script_dir}/trustee"
+}
+
+provide_image_key() {
+	kubectl exec -n "${kbs_namespace}" "${kbs_pod}" -- mkdir -p "/opt/confidential-containers/kbs/repository/$(dirname "$key_path")"
+	cat "${key_file}" | kubectl exec -i -n "${kbs_namespace}" "${kbs_pod}" -- tee "/opt/confidential-containers/kbs/repository/${key_path}" > /dev/null
+}
+
+launch_pod() {
+	kubectl apply -f "${script_dir}/nginx-encrypted.yaml"
+	export pod_name=$(kubectl -n default get pods -o NAME)
+}
+
+delete_pod() {
+	kubectl delete "${pod_name}"
+}
+
+check_image_key() {
+	kubectl logs -n "${kbs_namespace}" "${kbs_pod}" | grep "${key_path}"
+}
+
+set_metadata_annotation() {
+	local yaml="${1}"
+	local key="${2}"
+	local value="${3}"
+	local metadata_path="${4:-}"
+	local annotation_key=""
+
+ 	[ -n "$metadata_path" ] && annotation_key+="${metadata_path}."
+
+ 	# yaml annotation key name.
+ 	annotation_key+="metadata.annotations.\"${key}\""
+
+ 	echo "$annotation_key"
+	# yq set annotations in yaml. Quoting the key because it can have
+	# dots.
+	yq -i ".${annotation_key} = \"${value}\"" "${yaml}"
+}
+
+set_aa_kbc() {
```

Comment:
The added kernel_params metadata is not quoted:
```
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: nginx
  name: nginx-encrypted
  annotations:
    io.katacontainers.config.hypervisor.kernel_params: agent.guest_components_rest_api=resource agent.aa_kbc_params=cc_kbc::http://192.168.122.180:31231
spec:
  replicas: 1
  selector:
    matchLabels:
      app: nginx

<snip>
```

Don't know whether it will be a problem or not but we better quote it to avoid any parsing issues. Something like ` io.katacontainers.config.hypervisor.kernel_params: ' agent.guest_components_rest_api=resource agent.aa_kbc_params=cc_kbc::http://192.168.122.180:31231'`